src/ehttpc.erl

```diff
@@ -138,6 +149,13 @@ request(Worker, Method, Request, Timeout, Retry) when is_pid(Worker) ->
             {error, {ehttpc_worker_down, Reason}}
     end.
 
+%% @doc Send an async request. The callback is evaluated when an error happens or http response is received.
+-spec request_async(pid(), method(), request(), timeout(), callback()) -> ok.
+request_async(Worker, Method, Request, Timeout, ResultCallback) when is_pid(Worker) ->
+    ExpireAt = now_() + Timeout,
+    _ = erlang:send(Worker, ?ASYNC_REQ(Method, Request, ExpireAt, ResultCallback)),
```

Comment:
so we don't have to receive the `'$gen_cast'` message.

Reply:
The receive loop optimisation forced us to match `'$gen_call'` message, although using a macro like below:
`-define(GEN_CALL_REQ(From, Call), {'$gen_call', From, ?REQ(_, _, _) = Call}).`
This was due to compatibility reason. For this new API, I tried to avoid doing it again.